docs/snapshotting/snapshot-support.md

```diff
@@ -241,10 +241,69 @@ reason, the wall-clock should be updated to the current time, on the guest-side.
 More details on how you could do this can be found at a
 [related FAQ](../../FAQ.md#my-guest-wall-clock-is-drifting-how-can-i-fix-it).
 
-### Important notes
+### Provisioning host disk space for snapshots
 
 Depending on VM memory size, snapshots can consume a lot of disk space. Firecracker 
 integrators **must** ensure that the provisioned disk space is sufficient for normal
 operation of their service as well as during failure scenarios. If the service exposes
 the snapshot triggers to customers, integrators **must** enforce proper disk quotas to 
 avoid any DoS threats that would cause the service to fail or function abnormally.
+
+### Snapshot security and uniqueness
+
+When snapshots are used in a such a manner that a given guest's state is resumed
+from more than once, guest information assumed to be unique may in fact not be;
+this information can include identifiers, random numbers and random number
+seeds, as well as cryptographic tokens. Without a strong mechanism that enables
+users to guarantee that unique things stay unique across snapshot restores, we
+consider resuming execution from the same state more than once insecure.
+
+#### Usage examples
+
+##### Example 1: secure usage (currently in dev preview)
+
+```
+Boot microVM A -> ... -> Create snapshot S -> Terminate
+                                           -> Load S in microVM B -> Resume -> ...
+```
+
+Here, microVM A terminates after creating the snapshot without ever resuming
+work, and a single microVM B resumes execution from snapshot S. In this case,
+unique identifiers, random numbers, and cryptographic tokens that are meant to
+be used once are indeed only used once.
+
+##### Example 2: potentially insecure usage
+
+```
+Boot microVM A -> ... -> Create snapshot S -> Resume -> ...
+                                           -> Load S in microVM B -> Resume -> ...
+```
+
+Here, both microVM A and B do work staring from the state stored in snapshot S.
+Unique identifiers, random numbers, and cryptographic tokens that are meant to
+be used once may be used twice. It doesn't matter if microVM A is terminated
```

Comment:
I think this needs more detail about the risks of duplicating cryptographic tokens/entropy and how this lowers the security posture or can break a system in the case of duplicated unique identifiers.

Reply:
Makes sense, I can add one example for a uniqueness issue (e.g., duplicated identifier committed to a database), and one for a reuse of a random seed.

Reply:
I ended up adding a statement that this type of usage is insecure, since adding more clear examples is quite tricky and would need a much more extensive list.